### PR TITLE
squid - updated to latest version 5.3

### PR DESCRIPTION
**Pull request by fritzkink**

### CODE REVIEW

**Comment by AndWac** [comment]
This looks like a missing dependency:
/export/home/andreas/oi-userland/components/web/squid/build/manifest-i386-squid.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.file=libtdb.so.1 \
        pkg.debug.depend.reason=usr/squid/libexec/ext_session_acl \
        pkg.debug.depend.type=elf \
        pkg.debug.depend.path=lib/64 \
        pkg.debug.depend.path=usr/gcc/7/lib/amd64 \
        pkg.debug.depend.path=usr/lib/64'.
gmake: *** [/export/home/andreas/oi-userland/make-rules/ips.mk:502: /export/home/andreas/oi-userland/components/web/squid/build/.resolved-i386] Error 1

library/tdb is installed on our build server.

**Comment by fritzkink** [comment]
That's interesting. I havent installed library/tdb so no dependency to 
it. But after installing library/tdb a dependency was created. And the 
funny part: the package itself did not change at all.

Am 23.01.2022 um 15:58 schrieb Andreas Wacknitz:

File: components/web/squid/Makefile [hunk 1]
```diff
@@ -95,50 +95,45 @@ CONFIGURE_OPTIONS += --enable-referer-log
...
 CONFIGURE_OPTIONS += --enable-ssl
+CONFIGURE_OPTIONS += --enable-ssl-crtd
+#CONFIGURE_OPTIONS += --enable-devpoll
+CONFIGURE_OPTIONS += --enable-zph-qos
+CONFIGURE_OPTIONS += --enable-icap-client
+CONFIGURE_OPTIONS += --with-openssl=$(OPENSSL_PREFIX)
 CONFIGURE_OPTIONS += --enable-storeio='aufs,diskd,ufs'
+CONFIGURE_OPTIONS += --enable-storeid-rewrite-helpers=file
+CONFIGURE_OPTIONS += --with-default-user=webservd
+CONFIGURE_OPTIONS += --enable-inline
```
Comment (AndWac) [inline components/web/squid/Makefile:105]: --with-default-user=webservd is already above

File: components/web/squid/Makefile [hunk 2]
```diff
@@ -95,50 +95,45 @@ CONFIGURE_OPTIONS += --enable-referer-log
...
 CONFIGURE_OPTIONS += --enable-useragent-log
+CONFIGURE_OPTIONS += --enable-log-daemon-helpers='DB,file'
```
Comment (AndWac) [inline components/web/squid/Makefile:107]: You should add
--with-tdb
to make the use of tdb explicit.

File: components/web/squid/Makefile [hunk 3]
```diff
@@ -95,50 +95,44 @@ CONFIGURE_OPTIONS += --enable-referer-log
...
+CONFIGURE_OPTIONS += --enable-wccpv2
+# Native instructions need to be disabled for zones as reported in #5221
+# This is described in detail in
+#   http://wiki.squid-cache.org/KnowledgeBase/IllegalInstructionError
+CONFIGURE_OPTIONS += --disable-arch-native
 
 # common targets
 build:		$(BUILD_64) 
 
 install:	$(INSTALL_64)
```
Comment (AndWac) [inline components/web/squid/Makefile]: The common targets (build:, install:, and test:) can be removed  as they are automatically set to these values.

Comment (alarcher) [reply components/web/squid/Makefile]: For common Makefiles, you can run 'gmake update' in the component directory to apply some formatting and bump the component revision. I tried to implement the most common changes that we had to do in the past 2 years.

Comment (fritzkink) [reply components/web/squid/Makefile]: Thanks Aurélien, this make target was new to me. Updated Makefile accordingly.

File: components/web/squid/Makefile [hunk 4]
```diff
@@ -22,45 +22,45 @@
...
 include ../../../make-rules/shared-macros.mk
 
-PATH=/usr/bin:/usr/gnu/bin:/usr/sbin:/usr/perl5/bin
-
 COMPONENT_NAME=         squid
-COMPONENT_VERSION=      3.5.28
-COMPONENT_SUMMARY=	Squid Web Proxy Cache
+COMPONENT_VERSION=      5.3
+COMPONENT_REVISION=     1
+COMPONENT_SUMMARY=      Squid Web Proxy Cache
```
Comment (AndWac) [inline components/web/squid/Makefile:34]: COMPONENT_REVISION needs to be reset as you have changed the COMPONENT_VERSION and we didn't have released it yet.

**Pull request merged**